File: src/test/scala/com/github/tetrisanalyzer/piecegenerator/JavaPieceGeneratorTest.scala
package com.github.tetrisanalyzer.piecegenerator

import com.github.tetrisanalyzer.BaseTest
import org.junit.Test

class JavaPieceGeneratorTest extends BaseTest {
  @Test def nextPiece() {
    val pieceGenerator = new JavaPieceGenerator

    for (i <- 1 to 1000) {
      val pieceNumber = pieceGenerator.nextPiece.number.toInt
      pieceNumber should be >= 1
      pieceNumber should be <= 7
    }
  }
}

File: langs/java/src/test/java/com/github/tetrisanalyzer/piecegenerator/JavaPieceGeneratorTest.java
package com.github.tetrisanalyzer.piecegenerator;

import com.github.tetrisanalyzer.settings.TetrisAnalyzerGameSettings;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class JavaPieceGeneratorTest {

    @Test
    public void nextPiece() {
        PieceGenerator pieceGenerator = new JavaPieceGenerator(new TetrisAnalyzerGameSettings());

        for (int i=1; i<=1000; i++) {
            int pieceNumber = pieceGenerator.nextPiece().number();

            assertTrue(pieceNumber >= 1);
            assertTrue(pieceNumber <= 7);
        }
    }
}
